### PR TITLE
Added flows for generating sparsezoo structure to ModuleExporter

### PR DESCRIPTION
**Pull request by kevinaer**
- Added flow for exporting Pytorch Modules to SparseZoo structure.
- Small quality fixes

### CODE REVIEW

**Review by markurtz — CHANGES_REQUESTED** [review]
Synced with Kevin on this to make a few changes:
- make sure PIL dependency isn't going to fail for non CV flows
- look into the ModuleTrainer flows that have logic for extracting features and labels from a generic input and reusing here
- look into requiring a more generic iterable instance for the data and the user can optionally supply an iterable for the originals or a callback to recover the original 
- add in optional export of the entire model in a torch.save fashion rather than just saving the state_dict

File: src/sparseml/pytorch/utils/exporter.py [hunk 1]
```diff
@@ -154,13 +238,38 @@ def export_onnx(
             _delete_trivial_onnx_adds(onnx_model)
             onnx.save(onnx_model, onnx_path)
 
+    def export_torchscript(
+        self,
+        name: str = "model.pts",
+        trace: bool = False,
+        sample_batch: Optional[Any] = None,
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:245]: nit: can we remove `trace`, and just run a trace if a `sample_batch` is provided?

**Review by bfineran — APPROVED** [review]
LGTM

File: src/sparseml/pytorch/utils/exporter.py [hunk 2]
```diff
@@ -154,13 +241,37 @@ def export_onnx(
...
+        Otherwise uses script to create torchscript.
+
+        :param name: name of the torchscript file to save
+        :param sample_batch: If provided, will create torchscript model via tracing
+            using the sample_batch
+        """
+        path = os.path.join(self._output_dir, "framework", name)
+        create_parent_dirs(path)
+        if sample_batch:
+            if sample_batch is None:
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:261]: nit: unreachable line

**Review by bfineran — CHANGES_REQUESTED** [review]

**Review by bfineran — APPROVED** [review]
LGTM

**Review by markurtz — APPROVED** [review]
